distributed/worker.py

```diff
@@ -1284,13 +1308,14 @@ async def close(
                 with suppress(TimeoutError):
                     await self.batched_stream.close(timedelta(seconds=timeout))
 
-            self.actor_executor._work_queue.queue.clear()
-            if isinstance(self.executor, ThreadPoolExecutor):
-                self.executor._work_queue.queue.clear()
-                self.executor.shutdown(wait=executor_wait, timeout=timeout)
-            else:
-                self.executor.shutdown(wait=False)
-            self.actor_executor.shutdown(wait=executor_wait, timeout=timeout)
+            for executor in self.executors.values():
+                if executor is utils._offload_executor:
+                    continue  # Never shutdown the offload executor
+                if isinstance(executor, ThreadPoolExecutor):
+                    executor._work_queue.queue.clear()
+                    executor.shutdown(wait=executor_wait, timeout=timeout)
+                else:
+                    executor.shutdown(wait=False)
```

Comment:
Should we also pass `wait=executor_wait` and `timeout=timeout` keywords here like we do above? For example, I might have a custom `ProcessPoolExecutor` I'm using and want to have `wait=True`.

Reply:
Added `wait=executor_wait` but `timeout=timeout` isn't a valid `concurrent.futures.Executor` argument. It is introduced in `ThreadPoolExecutor`.